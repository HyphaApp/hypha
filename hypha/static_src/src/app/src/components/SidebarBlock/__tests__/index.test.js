import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import SidebarBlock from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

describe("Test Side bar component", () => {
    const title = "title text"

    const subject = mount(<SidebarBlock title={title}>
      <span>Children</span></SidebarBlock>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.sidebar-block').length).toBe(1);
      expect(subject.find('h5').text()).toBe(title);
      expect(subject.find('div').children().length).toBe(4);
      expect(subject.find('span').text()).toBe("Children");
    });
  
    test('render a side bar component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
