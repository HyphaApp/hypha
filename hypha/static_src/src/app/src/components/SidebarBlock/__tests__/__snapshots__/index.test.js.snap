// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Side bar component render a side bar component 1`] = `
<SidebarBlock
  title="title text"
>
  <div
    className="sidebar-block"
  >
    <h5>
      title text
    </h5>
    <span>
      Children
    </span>
    <div
      className="partition"
    >
      Â 
    </div>
  </div>
</SidebarBlock>
`;
